torch_geometric/utils/homophily.py

```diff
@@ -63,11 +85,26 @@ def homophily(edge_index: Adj, y: Tensor, batch: OptTensor = None,
         else:
             return scatter_mean(out, batch[col], dim=0)
 
-    else:
+    elif method == 'node':
         out = torch.zeros(row.size(0), device=row.device)
         out[y[row] == y[col]] = 1.
         out = scatter_mean(out, col, 0, dim_size=y.size(0))
         if batch is None:
             return float(out.mean())
         else:
             return scatter_mean(out, batch, dim=0)
+
+    else:
```

Comment:
The other homophily methods support a `batch` argument. I haven't thought how, but it would be good to support that for this too.

Reply:
Okay, that makes sense. I will add it on the weekend

Reply:
Added `batch` support.